File: src/main/java/mnm/mods/tabbychat/api/filters/FilterEvent.java
package mnm.mods.tabbychat.api.filters;

import java.util.Set;
import java.util.regex.Matcher;

import com.google.common.collect.Sets;

import mnm.mods.tabbychat.api.Channel;
import net.minecraft.util.text.ITextComponent;

public class FilterEvent {

    public final Matcher matcher;
    public ITextComponent text;
    public Set<Channel> channels = Sets.newHashSet();

    public FilterEvent(Matcher matcher, Set<Channel> channels, ITextComponent text) {
        this.matcher = matcher;
        this.text = text;
        this.channels = channels;
    }
}

File: src/main/kotlin/mnm/mods/tabbychat/api/filters/FilterEvent.kt
package mnm.mods.tabbychat.api.filters

import mnm.mods.tabbychat.api.Channel
import net.minecraft.util.text.ITextComponent
import java.util.regex.Matcher

class FilterEvent(val matcher: Matcher, var channels: MutableSet<Channel>, var text: ITextComponent)
